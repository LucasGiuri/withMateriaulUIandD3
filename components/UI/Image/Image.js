import PropTypes from 'prop-types';
import { Img } from './Image.styles';

const Image = ({isLogo, url, alt, onClick}) => (
  <Img src={url} alt={alt} isLogo={isLogo} onClick={onClick} />
);

Image.propTypes = {
  isLogo: PropTypes.bool,
  url: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

Image.defaultProps = {
  isLogo: false,
  onClick: () => {}
};

export default Image;