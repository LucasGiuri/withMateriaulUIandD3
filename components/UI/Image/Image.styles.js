import styled from 'styled-components';

export const Img = styled.img`
  padding-top: 8px;
  padding-bottom: 8px;
  width: ${props => props.isLogo ? '100px': '200px'};
`;