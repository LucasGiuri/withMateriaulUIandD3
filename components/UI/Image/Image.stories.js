import React from 'react';
import { storiesOf } from '@storybook/react';
import Image from './Image';

const url = 'http://pokestadium.com/sprites/xy/charizard.gif';
const alt = 'alt';

storiesOf('Image')
  .add('Image default', () => (
    <Image url={url} alt={alt} />
  ))
  
  .add('Image with isLogo attr', () => (
    <Image isLogo={true} url={url} alt={alt} />
  ))