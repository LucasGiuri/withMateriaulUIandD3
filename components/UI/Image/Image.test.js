import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image';
import 'jest-styled-components';

describe('Image', () => {
  it('Image basic render snapshoot', () => {
    const url = 'http://pokestadium.com/sprites/xy/charizard.gif';
    const alt = 'alt';
    const component = renderer.create(<Image url={url} alt={alt} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Image isLogo render snapshoot', () => {
    const url = 'http://pokestadium.com/sprites/xy/charizard.gif';
    const alt = 'alt';
    const component = renderer.create(<Image url={url} alt={alt} isLogo={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Image onClick function is called', () => {
    const url = 'http://pokestadium.com/sprites/xy/charizard.gif';
    const alt = 'alt';
    const onClick = jest.fn();
    const component = renderer.create(<Image url={url} alt={alt} onClick={onClick} />);
    const link = component.root.findByType('img');
    link.props.onClick();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith();
  });
})
