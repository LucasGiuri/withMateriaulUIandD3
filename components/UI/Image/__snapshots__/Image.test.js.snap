// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Image basic render snapshoot 1`] = `
.c0 {
  padding-top: 8px;
  padding-bottom: 8px;
  width: 200px;
}

<img
  alt="alt"
  className="c0"
  onClick={[Function]}
  src="http://pokestadium.com/sprites/xy/charizard.gif"
/>
`;

exports[`Image Image isLogo render snapshoot 1`] = `
.c0 {
  padding-top: 8px;
  padding-bottom: 8px;
  width: 100px;
}

<img
  alt="alt"
  className="c0"
  onClick={[Function]}
  src="http://pokestadium.com/sprites/xy/charizard.gif"
/>
`;
